src/time-picker/time-picker.jsx

```diff
@@ -128,10 +130,19 @@ const TimePicker = React.createClass({
   },
 
   setTime(t) {
-    this.setState({
-      time: t,
-    });
-    this.refs.input.setValue(this.formatTime(t));
+    if (t){
+      this.setState({
+        time: t,
+      });
+      
+      this.refs.input.setValue(this.formatTime(t));
```

Comment:
Can you use a temporary variable to set the new time instead of duplicating the logic in the if/else ?